src/main/groovy/com/twcable/gradle/cqpackage/CqPackagePlugin.groovy

```diff
@@ -103,10 +89,35 @@ class CqPackagePlugin implements Plugin<Project> {
     }
 
 
+    @SuppressWarnings("GroovyAssignabilityCheck")
+    private void addDepConfiguration(Project project) {
+        project.logger.debug("Creating configuration: ${CQ_PACKAGE}")
+        def cqPackageConf = project.configurations.create CQ_PACKAGE
+
+        // attach to "runtime", but don't insist that it have to be there first (or will ever be there)
+        def runtimeConf = project.configurations.findByName("runtime")
```

Comment:
Not really...

FWIW, what you have wouldn't compile. It "should" be:
```groovy
project.plugins.withType(org.gradle.api.plugins.JavaPlugin) {
    project.configurations.findByName("runtime").with { Configuration conf ->
        cqPackageConf.extendsFrom(conf)
    }
}
```
which is functionally no different than:
```groovy
if (project.plugins.findPlugin(org.gradle.api.plugins.JavaPlugin) != null) {
    Configuration conf = project.configurations.findByName("runtime")
    cqPackageConf.extendsFrom(conf)
}
```
Which would fail to have `cq_package` extend from `runtime` if the `build.gradle` looks like
```groovy
apply plugin: "cqpackage"
apply plugin: "java"
```
because the "java" plugin will be applied **after** this.

What I think you meant was
```groovy
project.plugins.whenPluginAdded { Plugin plugin ->
    if (plugin instanceof org.gradle.api.plugins.JavaPlugin) {
        Configuration conf = project.configurations.findByName("runtime")
        cqPackageConf.extendsFrom(conf)
    }
}
```
however that wouldn't work if the `build.gradle` looks like
```groovy
apply plugin: "java"
apply plugin: "cqpackage"
```
because the "java" plugin wold have already been applied before the listener was set up.

So what would really be needed is
```groovy
if (project.plugins.findPlugin(org.gradle.api.plugins.JavaPlugin) != null) {
    Configuration conf = project.configurations.findByName("runtime")
    cqPackageConf.extendsFrom(conf)
}
else {
    project.plugins.whenPluginAdded { Plugin plugin ->
        if (plugin instanceof org.gradle.api.plugins.JavaPlugin) {
            Configuration conf = project.configurations.findByName("runtime")
            cqPackageConf.extendsFrom(conf)
        }
    }
}
```
to cover both the cases of the plugin already being there and listening for it to appear later.